### PR TITLE
Add go best practices references

### PR DESCRIPTION
**Pull request by macleodbroad-wf**
R: @chrislococo-wf @nickmiller-wf 

@Workiva/devtools-pp

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Raven
Number of Findings: 0

File: go/style.md [hunk 1]
```diff
@@ -30,6 +30,6 @@ Performant Go
...
-2.  Prefer fewer channels. Justification: Channels are slow and it’s nearly impossible to guarantee you won’t leave zombie goroutines lying around. Channels are threadsafe queues; use them where you might use a queue. Don’t shoehorn them in just because they exist. If you need a high-performance queue, there are alternatives.
+2.  Prefer fewer channels. Justification: Channels [are slow](http://w-dev-blog.appspot.com/posts/2015/03/05/performant-concurrent-queue/index.html) and it’s nearly impossible to guarantee you won’t leave [zombie goroutines](https://groups.google.com/forum/#!topic/golang-nuts/uiySuH8_3Y4) lying around. Channels are threadsafe queues; use them where you might use a queue. Don’t shoehorn them in just because they exist. If you need a high-performance queue, there are [alternatives](https://github.com/Workiva/go-datastructures/blob/master/queue/queue.go).
```
Comment (nickmiller-wf) [inline go/style.md]: I think the dev blog link is only visible to Workiva devs. Is that ok even though this is intended to be a public repo?

Comment (macleodbroad-wf) [inline go/style.md]: That's true. I think it is OK as-is for now, and we'll circle back to it when/if we make that blog post public.

Would you prefer I remove it for now? It's still a good reference for internal people.

Comment (nickmiller-wf) [inline go/style.md]: Makes sense to leave it for now

**Comment by nickmiller-wf** [comment]
+1

**Comment by chrislococo-wf** [comment]
+1

**Pull request merged**